remove webCryptoMasterKeyFetch from UI process main-thread

nmahendru opened the pull request:
#### 795f182ee82bb8f6d85121624aa7423a518341f3
<pre>
remove webCryptoMasterKeyFetch from UI process main-thread
<a href="https://bugs.webkit.org/show_bug.cgi?id=275978">https://bugs.webkit.org/show_bug.cgi?id=275978</a>
<a href="https://rdar.apple.com/129814280">rdar://129814280</a>

Reviewed by Sihui Liu and Alex Christensen.

Add a new Async selector for fetching the webCryptoMasterKey in NavigationClient.
Replace the existing sync selector with an Async one in DataStoreDelegate.

This lets WebKit clients(Safari, for instance) have the option to fetch the master key off the
main-thread. The existing sync flow for fetching the key still works but has been marked
as legacy and will be removed in the future.

The DataStore async selector will now be used in all the Service worker tests and thus will be
tested.

* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/APINavigationClient.h:
(API::NavigationClient::legacyWebCryptoMasterKey):
(API::NavigationClient::webCryptoMasterKey): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageNavigationClient):
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebsiteDataStoreDelegate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::setNavigationDelegate):
(WebKit::NavigationState::NavigationClient::legacyWebCryptoMasterKey):
(WebKit::NavigationState::NavigationClient::webCryptoMasterKey): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::getWebCryptoMasterKey):
(WebKit::WebPageProxy::wrapCryptoKey):
(WebKit::WebPageProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::getWebCryptoMasterKey):
(WebKit::WebProcessProxy::wrapCryptoKey):
(WebKit::WebProcessProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreClient.cpp: Copied from Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.h.
(WebKit::WebsiteDataStoreClient::webCryptoMasterKey):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreClient.h:
(WebKit::WebsiteDataStoreClient::webCryptoMasterKey): Deleted.
(WebKit::WebsiteDataStoreClient::webCryptoMasterKeyAsync): Deleted.
(WebKit::WebsiteDataStoreClient::isAsyncMasterKeyDelegateSet): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.h:
* Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.mm:
(-[TestWebsiteDataStoreDelegate webCryptoMasterKey:]):
(-[TestWebsiteDataStoreDelegate getWebCryptoMasterKey:]): Deleted.
(-[TestWebsiteDataStoreDelegate webCryptoMasterKey]): Deleted.

Canonical link: <a href="https://commits.webkit.org/280651@main">https://commits.webkit.org/280651@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1f61d5115f584791f5e87c0122e64f9eed1230bd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57214 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36542 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9689 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60835 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7657 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59342 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44166 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7847 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46322 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5390 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59244 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34289 "Found 1 new test failure: compositing/geometry/video-opacity-overlay.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49398 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27182 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31071 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6714 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6662 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/53038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6984 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62515 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1127 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/7079 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53583 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1132 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49437 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/53655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/947 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8534 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32371 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33456 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34541 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33202 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebsiteDataStoreDelegate.h:
```diff
@@ -72,5 +72,6 @@ WK_API_AVAILABLE(macos(10.15), ios(13.0))
 - (void)websiteDataStore:(WKWebsiteDataStore *)dataStore domain:(NSString *)registrableDomain didOpenDomainViaWindowOpen:(NSString *)openedRegistrableDomain withProperty:(WKWindowProxyProperty)property directly:(BOOL)directly;
 - (void)websiteDataStore:(WKWebsiteDataStore *)dataStore didAllowPrivateTokenUsageByThirdPartyForTesting:(BOOL)wasAllowed forResourceURL:(NSURL *)resourceURL;
 - (void)websiteDataStore:(WKWebsiteDataStore *)dataStore domain:(NSString *)registrableDomain didExceedMemoryFootprintThreshold:(size_t)footprint withPageCount:(NSUInteger)pageCount processLifetime:(NSTimeInterval)processLifetime inForeground:(BOOL)inForeground wasPrivateRelayed:(BOOL)wasPrivateRelayed canSuspend:(BOOL)canSuspend;
+- (void)getWebCryptoMasterKey:(void (^)(NSData *))completionHandler;
```
We could probably call this `webCryptoMasterKeyAsync`

nmahendru replied:
This name was chose as per feedback on the safari PR.

szewai commented on Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
```diff
@@ -86,21 +86,30 @@
 private:
 
-    std::optional<Vector<uint8_t>> webCryptoMasterKey() final
+    void webCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) final
```
This change will make existing `webCryptoMasterKey` unused. 
For consistency, we should either drop the delegate method `webCryptoMasterKey`, or add a new `m_hasWebCryptoMasterKeyAsyncSelector` and make `void webCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) ` check both delegate methods (otherwise exiting clients might be broken).

achristensen07 replied:
It would maximize compatibility to call the old selector if present and call the CompletionHandler with the result synchronously then return, and then try calling the new selector if present.

nmahendru replied:
webCryptoMasterKey is still used by WebKitLegacy so I have left it there.

nmahendru replied:
I think I spoke to soon and totally misunderstood your suggestion. I am gonna keep both as you suggested @szewai  and try the async one first and then the sync one for existing clients.

achristensen07 commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2547,38 +2547,42 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
-        if (wrapSerializedCryptoKey(*masterKey, key, wrappedKey)) {
-            completionHandler(WTFMove(wrappedKey));
-            return;
+    getWebCryptoMasterKey([&key, completionHandler = WTFMove(completionHandler)](std::optional<Vector<uint8_t>> && masterKey) mutable {
```
Make key a Vector&& and move it to the capture.  Otherwise it's likely a UAF.

achristensen07 commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2547,38 +2547,42 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
-        if (auto key = WebCore::unwrapCryptoKey(*masterKey, wrappedKey)) {
-            completionHandler(WTFMove(key));
-            return;
+    getWebCryptoMasterKey([&wrappedKey, completionHandler = WTFMove(completionHandler)](std::optional<Vector<uint8_t>> && masterKey) mutable {
```
Ditto.  Same in WebPageProxy.

szewai reviewed (commented):
Some thoughts: 
1. To let client specify crypto key asynchronously, we only need one new method. It would be redundant to add two.
2. The new delegate method should be added to `_WKWebsiteDataStoreDelegate`, not `WKNavigationDelegate`, as we expect `WKWebView` using the same data store to have the same crypto key getter.
3. The existing delegate methods should still work for compatibility, which means there is a bug if the existing tests fail.
4. For simplicity, I think we could keep most of existing implementation by adding check for new delegate method after the existing check, e.g.:
```
void WebPageProxy::getWebCryptoMasterKey(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
{
    if (auto keyData = m_websiteDataStore->client().webCryptoMasterKey())
        return completionHandler(keyData);

    if (auto keyData = m_navigationClient->webCryptoMasterKey(*this))
        return completionHandler(Vector(keyData->span()));

    return m_websiteDataStore->client().webCryptoMasterKeyAsync(WTFMove(completionHandler));
}
```
and adding a new `webCryptoMasterKeyAsync` to `WebsiteDataStoreClient` without chaining existing `webCryptoMasterKey`.

szewai commented on Source/WebKit/UIProcess/API/APINavigationClient.h:
```diff
@@ -116,7 +116,7 @@ class NavigationClient {
     virtual void processDidBecomeUnresponsive(WebKit::WebPageProxy&) { }
 
-    virtual RefPtr<Data> webCryptoMasterKey(WebKit::WebPageProxy&) { return nullptr; }
+    virtual RefPtr<Data> webCryptoMasterKey(WebKit::WebPageProxy&, bool = false) { return nullptr; }
```
Let's use enum class for readability, like `enum class UseDefaultMasterKeyAsFallback :  bool { No, Yes };`

szewai commented on Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
```diff
@@ -87,10 +87,15 @@
     }
 
 private:
+    bool isAsyncMasterKeyDelegateSet() final
```
`canGetWebCryptoMasterKeyAsynchronously`

szewai commented on Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
```diff
@@ -103,6 +108,26 @@
 
+    void webCryptoMasterKeyAsync(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) final
+    {
+        if (m_hasWebCryptoMasterKeyAsyncSelector && m_delegate) {
```
We could return early for the null case since the non-null case is more complicated, e.g.
```
if (!m_hasWebCryptoMasterKeyAsyncSelector || !m_delegate)
    return completionHandler(std::nullopt);

auto checker = WebKit::CompletionHandlerCallChecker::create(m_delegate.getAutoreleased(), @selector(getWebCryptoMasterKey:));
...
```

szewai commented on Source/WebKit/UIProcess/Cocoa/NavigationState.h:
```diff
@@ -137,7 +137,7 @@ class NavigationState final : public PageLoadState::Observer {
         void processDidBecomeUnresponsive(WebPageProxy&) override;
 
-        RefPtr<API::Data> webCryptoMasterKey(WebPageProxy&) override;
+        RefPtr<API::Data> webCryptoMasterKey(WebPageProxy&, bool isAsyncMasterKeyDelegateSet = false) override;
```
Let's use `UseDefaultMasterKeyAsFallback` here

nmahendru replied:
updated.

szewai commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2556,38 +2557,48 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
-    if (auto isKey = defaultWebCryptoMasterKey())
-        return isKey;
-    return std::nullopt;
+    if (auto isKey = m_websiteDataStore->client().webCryptoMasterKey()) {
```
`isKey` => `key`?

nmahendru replied:
I would rather just stick to the `is` nomenclature that was there before.

szewai commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2556,38 +2557,48 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
+        else
+            completionHandler(std::nullopt);
+        return;
+    }
```
If we fall back to default key in `webCryptoMasterKeyAsync` when the new delegate method is not specified, then we don't need this check and we could simplify the change in NavigationState.mm. 
We might do it in a followup.

nmahendru replied:
sure.. lets get the Safari side done first.. this could be done later!

szewai reviewed (approved):
r=me if bots are happy

achristensen07 commented on Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.mm:
```diff
@@ -153,6 +153,14 @@ - (void)clearReportedWindowProxyAccessDomains
+    completionHandler([NSData dataWithBytes:keyBytes length:(keyLength)]);
+}
+
 - (NSData *)webCryptoMasterKey
```
There shouldn't be a reason to leave this version.  It should be never called, and support for the synchronous selector is on its way to being removed.

achristensen07 commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreClient.h:
```diff
@@ -126,6 +126,16 @@ class WebsiteDataStoreClient {
+        completionHandler(std::nullopt);
+    }
+
+    virtual bool isAsyncMasterKeyDelegateSet()
```
The existence of a selector shouldn't be exposed as a public method

achristensen07 commented on Source/WebKit/UIProcess/API/APINavigationClient.h:
```diff
@@ -116,7 +118,7 @@ class NavigationClient {
     virtual void processDidBecomeUnresponsive(WebKit::WebPageProxy&) { }
 
-    virtual RefPtr<Data> webCryptoMasterKey(WebKit::WebPageProxy&) { return nullptr; }
+    virtual RefPtr<Data> webCryptoMasterKey(WebKit::WebPageProxy&, DefaultMasterKeyAsFallback = DefaultMasterKeyAsFallback::No) { return nullptr; }
```
The parameter DefaultMasterKeyAsFallback makes the flow hard to follow.  I think this entire function should be removed in favor of only asking the WebsiteDataStoreClient.

achristensen07 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
```diff
@@ -103,6 +108,26 @@
         return makeVector(result.get());
     }
 
+    void webCryptoMasterKeyAsync(CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler) final
```
This should be just a part of webCryptoMasterKey.  If we want to keep supporting the synchronous selector, then call the synchronous selector and call the completion handler immediately with its result.

achristensen07 replied:
UIDelegate::UIClient::createNewPage is an example of a single C++ call that calls one of several selectors, one async and one sync.

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.h:
```diff
@@ -1767,8 +1767,8 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
-    void wrapCryptoKey(const Vector<uint8_t>&, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&&);
-    void unwrapCryptoKey(const struct WebCore::WrappedCryptoKey&, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&&);
+    void wrapCryptoKey(Vector<uint8_t>&&, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&&);
+    void unwrapCryptoKey(struct WebCore::WrappedCryptoKey &&, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&&);
```
nit: space before && should be removed.

achristensen07 commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2556,38 +2557,48 @@ void WebProcessProxy::getNotifications(const URL& registrationURL, const String&
 }
 
-void WebProcessProxy::unwrapCryptoKey(const struct WrappedCryptoKey& wrappedKey, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
+void WebProcessProxy::unwrapCryptoKey(struct WrappedCryptoKey&& wrappedKey, CompletionHandler<void(std::optional<Vector<uint8_t>>&&)>&& completionHandler)
```
I think "struct" is unnecessary.

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -11555,42 +11556,36 @@ void WebPageProxy::setOverlayScrollbarStyle(std::optional<WebCore::ScrollbarOver
+    if (m_navigationClient->legacyWebCryptoMasterKey(*this, WTFMove(completionHandler)))
+        return;
+
+    m_websiteDataStore->client().webCryptoMasterKey(WTFMove(completionHandler));
```
Use-after-move on `completionHandler`?

nmahendru replied:
Had a chat with @achristensen07, and he believes that although syntactically it is a UAM, but the function will early return if the completionHandler is called inside `legacyWebCryptoMasterKey`.

Regardless, I have refactored it out and made it nested so that now the UAM does not exist.

achristensen07 commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -11575,42 +11576,38 @@ void WebPageProxy::setOverlayScrollbarStyle(std::optional<WebCore::ScrollbarOver
-    if (auto keyData = m_navigationClient->webCryptoMasterKey(*this))
-        return Vector(keyData->span());
-    return std::nullopt;
+    m_navigationClient->legacyWebCryptoMasterKey(*this, [completionHandler = WTFMove(completionHandler), this](std::optional<Vector<uint8_t>>&& key) mutable {
```
We need protectedThis = Ref { *this } or else use of this could be a UAF.  Or a weakThis with a null check instead of capturing this.

nmahendru committed:
remove webCryptoMasterKeyFetch from UI process main-thread
https://bugs.webkit.org/show_bug.cgi?id=275978
rdar://129814280

Reviewed by Sihui Liu and Alex Christensen.

Add a new Async selector for fetching the webCryptoMasterKey in NavigationClient.
Replace the existing sync selector with an Async one in DataStoreDelegate.

This lets WebKit clients(Safari, for instance) have the option to fetch the master key off the
main-thread. The existing sync flow for fetching the key still works but has been marked
as legacy and will be removed in the future.

The DataStore async selector will now be used in all the Service worker tests and thus will be
tested.

* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/APINavigationClient.h:
(API::NavigationClient::legacyWebCryptoMasterKey):
(API::NavigationClient::webCryptoMasterKey): Deleted.
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetPageNavigationClient):
* Source/WebKit/UIProcess/API/Cocoa/WKNavigationDelegatePrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebsiteDataStoreDelegate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::setNavigationDelegate):
(WebKit::NavigationState::NavigationClient::legacyWebCryptoMasterKey):
(WebKit::NavigationState::NavigationClient::webCryptoMasterKey): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::getWebCryptoMasterKey):
(WebKit::WebPageProxy::wrapCryptoKey):
(WebKit::WebPageProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::getWebCryptoMasterKey):
(WebKit::WebProcessProxy::wrapCryptoKey):
(WebKit::WebProcessProxy::unwrapCryptoKey):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreClient.cpp: Copied from Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.h.
(WebKit::WebsiteDataStoreClient::webCryptoMasterKey):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreClient.h:
(WebKit::WebsiteDataStoreClient::webCryptoMasterKey): Deleted.
(WebKit::WebsiteDataStoreClient::webCryptoMasterKeyAsync): Deleted.
(WebKit::WebsiteDataStoreClient::isAsyncMasterKeyDelegateSet): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.h:
* Tools/WebKitTestRunner/cocoa/TestWebsiteDataStoreDelegate.mm:
(-[TestWebsiteDataStoreDelegate webCryptoMasterKey:]):
(-[TestWebsiteDataStoreDelegate getWebCryptoMasterKey:]): Deleted.
(-[TestWebsiteDataStoreDelegate webCryptoMasterKey]): Deleted.

Canonical link: https://commits.webkit.org/280651@main
